homeassistant/components/sensor/yweather.py

```diff
@@ -131,9 +131,12 @@ def entity_picture(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
-        return {
-            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
-        }
+        attrs = {ATTR_ATTRIBUTION: CONF_ATTRIBUTION}
+
+        if self._code is None or "weather" in self._type:
```

Comment:
Should the attribute added always (`attrs['condition_code'] = None`) even if "weather" isn't in `self._type`?

Reply:
No. Only the weather condition version of this sensor should have condition code since it doesn’t make much sense for the temp sensor to have it. 

This follows the pattern of `entity_image`

Reply:
If you want to follow that pattern, you should check for `is not None`

Reply:
Yeah, I blew this, copying the pattern for returning None lol. Late night PRs. 

I’ll fix it later today. Thanks.